scope_extensions-explainer.md

```diff
@@ -1,126 +1,216 @@
-
-Each object in `scope_extension` must contain both `type` and `value` string
-fields. The value of `type` must be one of `["origin" | "registrable_domain"]`.
-The `value` field must contain a valid URL. 
+The "Example" app has a regular scope of `http://example.com/app` and is
+extending its app scope to the origins `https://example.co.uk` and
+`https://help.example.com`.
 
-| type   | Behavior |
-|--------|----------|
-| `origin` | The URL in `value` is converted to an [origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-tuple). The `/` path of that origin is added to the extended scope.|
-| `registrable_domain` | The URL in `value` is converted to a [registrable domain](https://url.spec.whatwg.org/#host-registrable-domain). The `/` paths of all sub-domain origins within the registrable domain are added to the extended scope.|
-
-This format allows for both backward and forward compatibility. For example, a
-new type of entry could filter the paths of an origin added to the extended
-scope. Entries with types are that not recognized by a user agent should be
-ignored.
+* Each entry in `scope_extensions` must contain both `type` and `value` string
+fields.
+* `type` must be `"origin"`. Other types could be added in the future.
+* `value` must a valid URL. The URL is converted to an
+  [origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-tuple). 
 
 ### Association file
- Specify a `web-app-origin-association` file that must be located at
- `https://<associated origin>/.well-known/web-app-origin-association` on the
- associated origin's domain. This specifies a list of web apps that may include
- it as a scope extension.
 
-   Example association file located at
+A `web-app-origin-association` file must be served from
+`https://<associatedorigin>/.well-known/web-app-origin-association`. An app is
+allowed to extend its scope to this origin if their manifest ID is found in this
+file.
+
+Example association file located at
    `https://example.co.uk/.well-known/web-app-origin-association`:
-   ```json
-   {
-     "web_apps": [{
-       "web_app_identity": "https://example.com/"
-     }, {
-       "web_app_identity": "https://associated.site.com/"
-     }]
-   }
-   ```
 
-The `web_app_identity` field must contain a valid [web application
-id](https://w3c.github.io/manifest/#id-member).
+```json
+{
+  "https://example.com/app": {
+    "scope": "/app"   // Evaluates to https://example.co.uk/app
+  }
+}
+```
 
-### Resulting extended scope
-A URL is in the extended scope of a web app if both:
-  - It matches an entry in the `scope_extensions` list.
-  - That entry has been validated by fetching the
-    `<origin>/.well-known/web-app-origin-association` association file with an
-    entry matching the app's
-    [identity](https://w3c.github.io/manifest/#id-member).
+* Each dictionary key must be a validly formatted [web application
+id](https://w3c.github.io/manifest/#id-member).
```

Comment:
Nit: Web application ID has a very different meaning in Chromium. I would write manifest ID, which is what the link is pointing to, to disambiguate.

Reply:
Changed to "manifest id" throughout.